recipes/emsdk/all/conanfile.py

```diff
@@ -80,6 +88,12 @@ def package(self):
         tools.replace_in_file(toolchain,
                               "set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)",
                               "set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE BOTH)")
+        with tools.environment_append(self._emscripten_env):
+            tools.save("deleteme.cpp", "int main(){}")
+            self.run("em++ deleteme.cpp", run_environment=True) # force cache population
```

Comment:
This will break cross building.
It's okay for me to add this, but should be add an option to make this separation?

Reply:
indeed, This is most probably the cause of the failure on macos M1... It did not occur to me that this could be a problem, because this  recipe is already doing some kind of corss compilation from whatever to emscripten/wasm. I'm not used to Canadian Cross yet.